### ISSUE TITLE
Method 'get_Providers' in type 'Lucene.Net.Configuration.ConfigurationRoot' does not have an implementation.

### ISSUE THREAD

**Issue by JohnDuerden** [apache/lucenenet#323]
Upgrading from Beta 7 to Beta 11 I get _Method 'get_Providers' in type 'Lucene.Net.Configuration.ConfigurationRoot' does not have an implementation_ when adding a document to the index. It looks very similar to #311 but get_Providers not get_properties. Possibly I need to downgrade .Net Framework from 4.8 (Beta 7 ran OK on 4.8)?

**Comment by NightOwl888** [comment]
Thanks for the report.

What packages do you have from `Microsoft.Extensions.Configuration` installed in your project? And what versions?

I was able to reproduce this. I still need to look into why this is happening. The `Providers` property hasn't changed from `IEnumerable<IConfigurationProvider>`, so it is a bit odd that it is throwing an exception.

Fortunately, this time there is a workaround. You can bypass the `ConfigurationRoot` provider by injecting your own at application startup. Just provide your existing `IConfiguration` instance to `ConfigurationFactory`, or use the below code as the default.

```c#
IConfiguration configuration = new ConfigurationBuilder().AddEnvironmentVariables("lucene:").Build();

Lucene.Net.Configuration.ConfigurationSettings.SetConfigurationFactory(new ConfigurationFactory(configuration));
```

Here is one way to implement `ConfigurationFactory`. You could alternatively use a `Func<IConfiguration>` in the constructor and actively read it on the fly from its source if you need to.

```c#
    public class ConfigurationFactory : Lucene.Net.Configuration.IConfigurationFactory
    {
        private readonly IConfiguration configuration;

        public ConfigurationFactory(IConfiguration configuration)
        {
            this.configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
        }

        public IConfiguration GetConfiguration() => configuration;
    }
```

**Comment by JohnDuerden** [comment]
Thanks for the very prompt response. The work around worked fine. The Microsoft.Extensions.Configuration packages I had installed were  Microsoft.Extensions.Configuration.Abstractions and Microsoft.Extensions.Configuration.Binder. I have added for the workaround Microsoft.Extensions.Configuration.EnvironmentVariablers. The packages I had installed are prereqs to support the Microsoft.Extensions.Logging implementation.
